src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -80,6 +80,25 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, correlat
         confirmation_message = "I confirm I have an eligible Windows Server license with Azure Hybrid Benefit to apply this benefit to AKS on HCI or Windows Server. Visit https://aka.ms/ahb-aks for details"
         utils.user_confirmation(confirmation_message, yes)
 
+    if least_privilege is True:
+        # pre-checks specific to least privilege scenario
+        # check if config settings are passed
+        if config_settings is None:
+            telemetry.set_user_fault()
+            telemetry.set_exception(exception="Configuration settings are not passed", fault_type=consts.Config_Settings_Not_Passed_Least_Privileges_Fault_Type, summary="Configuration settings are not passed which are mandatory when onboarding with leastPrivileges")
+            raise ValidationError("Configuration settings are not passed", "Please pass required configuration settings using '--config-settings' flag while onboarding with least-privilege enabled.")
```

Comment:
should use ArgumentUsageError instead of ValidationError

Reply:
you should check out the error types supported by az cli: https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md
this file has some basic description of these error types: https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/azclierror.py

Reply:
also, setting ArgumentUsageError might already be setting it as user fault by default so we need not set it explicitly. But confirm this once